### PR TITLE
Fix Unreal crashing when deleting an actor with a JSBMovement component.

### PR DESCRIPTION
**Pull request by fadak0828**
Below is the crash log that occurred.

```
Unhandled Exception: EXCEPTION_ACCESS_VIOLATION reading address 0x00000000000000ac

UnrealEditor_JSBSimFlightDynamicsModelEditor!FJSBSimMovementCompVisualizer::DrawVisualization() [C:\workspace\jsbsim\UnrealEngine\Plugins\JSBSimFlightDynamicsModel\Source\JSBSimFlightDynamicsModelEditor\Private\JSBSimMovementCompVisualizer.cpp:55]
UnrealEditor_UnrealEd
UnrealEditor_UnrealEd
UnrealEditor_Renderer
UnrealEditor_Renderer
UnrealEditor_Renderer
UnrealEditor_Renderer
UnrealEditor_Renderer
UnrealEditor_UnrealEd
UnrealEditor_Engine
UnrealEditor_UnrealEd
UnrealEditor_UnrealEd
UnrealEditor_UnrealEd
UnrealEditor
UnrealEditor
UnrealEditor
UnrealEditor
UnrealEditor
UnrealEditor
kernel32
ntdll
```

### CODE REVIEW

**Comment by gallonmate** [comment]
Thanks. I can confirm that in UE5.4 this is broken.
I added a review/change request in your solution, because only one check is needed to prevent crashing. The three checks per function seem unnecessary and after looking at Unreal Engine source code, they also only use one check by using a single `if` at the beginning. For example:

```c++

void FSplineComponentVisualizer::DrawVisualization(const UActorComponent* Componen,.....
{
	if (const USplineComponent* SplineComp = Cast<const USplineComponent>(Component))
	{
               
        )    
```

File: UnrealEngine/Plugins/JSBSimFlightDynamicsModel/Source/JSBSimFlightDynamicsModelEditor/Private/JSBSimMovementCompVisualizer.cpp [hunk 1]
```diff
@@ -51,8 +51,22 @@ FJSBSimMovementCompVisualizer::~FJSBSimMovementCompVisualizer()
 
 void FJSBSimMovementCompVisualizer::DrawVisualization(const UActorComponent* Component, const FSceneView* View, FPrimitiveDrawInterface* PDI)
 {
+	if (!Component || !PDI)
+	{
+		return;
+	}
+
```
Comment (gallonmate) [inline UnrealEngine/Plugins/JSBSimFlightDynamicsModel/Source/JSBSimFlightDynamicsModelEditor/Private/JSBSimMovementCompVisualizer.cpp:58]: Not needed

File: UnrealEngine/Plugins/JSBSimFlightDynamicsModel/Source/JSBSimFlightDynamicsModelEditor/Private/JSBSimMovementCompVisualizer.cpp [hunk 2]
```diff
@@ -51,8 +51,22 @@ FJSBSimMovementCompVisualizer::~FJSBSimMovementCompVisualizer()
...
+	if (!MovementComponent)
+	{
+		return;
+	}
+
 	AActor* Owner = Component->GetOwner();
+	if (!Owner)
+	{
+		return;
+	}
```
Comment (gallonmate) [inline UnrealEngine/Plugins/JSBSimFlightDynamicsModel/Source/JSBSimFlightDynamicsModelEditor/Private/JSBSimMovementCompVisualizer.cpp:69]: Not needed

File: UnrealEngine/Plugins/JSBSimFlightDynamicsModel/Source/JSBSimFlightDynamicsModelEditor/Private/JSBSimMovementCompVisualizer.cpp [hunk 3]
```diff
@@ -100,8 +114,22 @@ void FJSBSimMovementCompVisualizer::DrawVisualization(const UActorComponent* Com
 
 void FJSBSimMovementCompVisualizer::DrawVisualizationHUD(const UActorComponent* Component, const FViewport* Viewport, const FSceneView* View, FCanvas* Canvas)
 {
+	if (!Component || !View || !Canvas)
+	{
+		return;
+	}
+
```
Comment (gallonmate) [inline UnrealEngine/Plugins/JSBSimFlightDynamicsModel/Source/JSBSimFlightDynamicsModelEditor/Private/JSBSimMovementCompVisualizer.cpp:121]: not needed

File: UnrealEngine/Plugins/JSBSimFlightDynamicsModel/Source/JSBSimFlightDynamicsModelEditor/Private/JSBSimMovementCompVisualizer.cpp [hunk 4]
```diff
@@ -100,8 +114,22 @@ void FJSBSimMovementCompVisualizer::DrawVisualization(const UActorComponent* Com
...
+	if (!MovementComponent)
+	{
+		return;
+	}
+
 	AActor* Owner = MovementComponent->GetOwner();
+	if (!Owner)
+	{
+		return;
+	}
```
Comment (gallonmate) [inline UnrealEngine/Plugins/JSBSimFlightDynamicsModel/Source/JSBSimFlightDynamicsModelEditor/Private/JSBSimMovementCompVisualizer.cpp:132]: not needed

**Review by gallonmate — CHANGES_REQUESTED** [review]
The `if (!movementComponent)` is good. The other checks can be removed.

**Review by gallonmate — APPROVED** [review]

**Comment by bcoconni** [comment]
Merged.
Thanks @fadak0828 for the PR.
Thanks @gallonmate for the review.

**Pull request merged**